src/parser.yy

```diff
@@ -180,8 +185,8 @@ void yyerror(bpftrace::Driver &driver, const char *s);
 %%
 
 program:
-                c_definitions config probes_and_subprogs END {
-                    driver.ctx.root = driver.ctx.make_node<ast::Program>($1, $2, std::move($3.second), std::move($3.first));
+                c_definitions config macros probes_and_subprogs END {
```

Comment:
Do we care where macros are defined? Meaning above/below config and c_definitions. I was thinking about this as well for global map declarations.

Reply:
I don't think we care. I'll leave it as-is for now, but we should definitely come by later and loosen restrictions for all constructs.